src/transfers/transferTransactionsFactory.ts

```diff
@@ -113,14 +122,17 @@ export class TransferTransactionsFactory {
             receiver: options.receiver,
         });
 
-        return new TransactionBuilder({
-            config: this.config!,
+        const transaction = new Transaction({
             sender: sender,
             receiver: receiver,
-            dataParts: dataParts,
-            gasLimit: extraGasForTransfer,
-            addDataMovementGas: true,
-        }).build();
+            chainID: this.config.chainID,
+            gasLimit: 0n,
+        });
+
+        this.setTransactionPayload(transaction, dataParts);
+        this.setGasLimit(transaction, undefined, extraGasForTransfer);
```

Comment:
Indeed. Under the hood, `setGasLimit` does the previous logic of `addDataMovementGas` when it reaches the fallback for `extraGasForTransfer`. Maybe write a comment here on the PR to confirm that all is good.

Reply:
Indeed, that is right.